dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -126,11 +127,8 @@ def _update_categories(self, df_series, prev_dependent_properties=None,
         :type df_series: pandas.DataFrame
         :return: None
         """
-        if hasattr(df_series, 'tolist'):
-            df_series = df_series.tolist()
 
-        self._categories = utils._combine_unique_sets(
-            self._categories, df_series)
+        self._categories.update(df_series.value_counts(dropna=False).to_dict())
```

Comment:
This like will replace the known values as opposed to updating counts:
I suggest utilizing the same functionality as in `__add__`, e.g.:
```python
category_count = df_series.value_counts(dropna=False).to_dict()
self._categories = utils.add_nested_dictionaries(self._categories, category_count)
```